itou/metabase/management/commands/_job_applications.py

```diff
@@ -64,89 +76,110 @@ def get_ja_time_spent_from_new_to_accepted_or_refused(ja):
-    },
-    {
-        "name": "date_candidature",
-        "type": "date",
-        "comment": "Date de la candidature",
-        "lambda": lambda o: o.created_at,
-    },
-    {
-        "name": "état",
-        "type": "varchar",
-        "comment": "Etat de la candidature",
-        "lambda": lambda o: get_choice(choices=JobApplicationWorkflow.STATE_CHOICES, key=o.state),
-    },
-    {
-        "name": "type",
-        "type": "varchar",
-        "comment": ("Type de la candidature (auto-prescription, candidature autonome, candidature via prescripteur)"),
-        "lambda": lambda o: get_choice(choices=SENDER_KIND_CHOICES, key=o.sender_kind),
-    },
-    {
-        "name": "sous_type",
-        "type": "varchar",
-        "comment": (
-            "Sous-type de la candidature (auto-prescription par EI, ACI..."
-            " candidature autonome, candidature via prescripteur PE, ML...)"
-        ),
-        "lambda": get_job_application_sub_type,
-    },
-    {
-        "name": "délai_prise_en_compte",
-        "type": "interval",
-        "comment": (
-            "Temps écoulé rétroactivement de état nouveau à état étude" " si la candidature est passée par ces états"
-        ),
-        "lambda": get_ja_time_spent_from_new_to_processing,
-    },
-    {
-        "name": "délai_de_réponse",
-        "type": "interval",
-        "comment": (
-            "Temps écoulé rétroactivement de état nouveau à état accepté"
-            " ou refusé si la candidature est passée par ces états"
-        ),
-        "lambda": get_ja_time_spent_from_new_to_accepted_or_refused,
-    },
-    {
-        "name": "motif_de_refus",
-        "type": "varchar",
-        "comment": "Motif de refus de la candidature",
-        "lambda": lambda o: get_choice(choices=JobApplication.REFUSAL_REASON_CHOICES, key=o.refusal_reason),
-    },
-    {
-        "name": "id_candidat_anonymisé",
-        "type": "varchar",
-        "comment": "ID anonymisé du candidat",
-        "lambda": lambda o: anonymize(o.job_seeker_id, salt="job_seeker.id"),
-    },
-    {
-        "name": "id_structure",
-        "type": "integer",
-        "comment": "ID de la structure destinaire de la candidature",
-        "lambda": lambda o: o.to_siae_id,
-    },
-    {
-        "name": "type_structure",
-        "type": "varchar",
-        "comment": "Type de la structure destinaire de la candidature",
-        "lambda": lambda o: o.to_siae.kind,
-    },
-    {
-        "name": "nom_structure",
-        "type": "varchar",
-        "comment": "Nom de la structure destinaire de la candidature",
-        "lambda": lambda o: o.to_siae.display_name,
-    },
-] + get_department_and_region_columns(
-    name_suffix="_structure",
-    comment_suffix=" de la structure destinaire de la candidature",
-    custom_lambda=lambda o: o.to_siae,
+def get_ja_hiring_date(ja):
+    logs = [log for log in ja.logs.all() if log.transition == JobApplicationWorkflow.TRANSITION_ACCEPT]
```

Comment:
```python
ja.logs.filter(to_state=ja._meta.model.state.STATE_ACCEPTED).first()
```

Reply:
Négatif chef. Ta proposition rajouterait des dizaines de millers de requêtes db.

Cf le commentaire récurrent `# We have to do all this in python to benefit from prefetch_related.`.

Seul `ja.logs.all()` est préloadé et est donc utilisable sans requête db supplémentaire.

Le moindre `ja.logs.filter(...)` donnerait lieu à de nouvelles requêtes db sans profiter du `prefetch_related`.

@kemar

Reply:
Ah bah le commentaire est pas au bon endroit alors :D

Reply:
Ok je l'ai rajouté à cet endroit. Avec cette logique il y a encore 10 autres endroits où je devrais l'ajouter mais je vais éviter.